dynamiqs/qarrays/qarray.py

```diff
@@ -654,6 +660,38 @@ def include_last_two_dims(axis: int | tuple[int, ...] | None, ndim: int) -> bool
     )
 
 
+def _is_key_in_batch_dims(key: int | slice | tuple, ndim: int) -> bool:
```

Comment:
This is not the correct typing for `key` (missing `None`, tuple type specification, and we can also have `x[1:2, (1, 2, 3), None]`).